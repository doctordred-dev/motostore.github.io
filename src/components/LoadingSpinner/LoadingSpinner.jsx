import React from 'react';
import styles from './LoadingSpinner.module.scss';

const LoadingSpinner = () => {
    return (
        <div className={styles.spinnerContainer}>
            <div className={styles.spinner}></div>
            <p className={styles.loadingText}>Завантаження...</p>
        </div>
    );
};

export default LoadingSpinner;
