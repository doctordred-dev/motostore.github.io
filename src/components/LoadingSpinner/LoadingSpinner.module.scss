.spinnerContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100%;
  background-color: #111;
}

.spinner {
  width: 50px;
  height: 50px;
  border: 5px solid #333;
  border-top: 5px solid #ff3c00;
  border-radius: 50%;
  animation: spin 1s linear infinite;
  margin-bottom: 20px;
}

.loadingText {
  color: #fff;
  font-size: 18px;
  font-weight: 500;
  margin-top: 15px;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}
