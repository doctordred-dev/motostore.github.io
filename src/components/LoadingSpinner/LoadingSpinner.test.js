import { render } from '@testing-library/react';
import LoadingSpinner from './LoadingSpinner';

describe('LoadingSpinner component', () => {
    test('matches snapshot', () => {
        const { asFragment } = render(<LoadingSpinner />);
        expect(asFragment()).toMatchSnapshot();
    });
});