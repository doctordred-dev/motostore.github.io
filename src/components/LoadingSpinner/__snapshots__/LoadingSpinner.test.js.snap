// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinnerContainer"
  >
    <div
      class="spinner"
    />
    <p
      class="loadingText"
    >
      Завантаження...
    </p>
  </div>
</DocumentFragment>
`;
